tests/functional/examples/test_hierarchical_data_structures_referencing_attributes.py

```diff
@@ -268,3 +269,62 @@ def test_referencing_attributes_with_rrel_all_in_one_splitstring():
 
     assert m.references[2].ref.name == 'x'
     assert m.references[2].ref == m.structs[0].vals[0]
+
+
+def test_referencing_attributes_with_rrel_and_full_path_access():
+    """
+    RREL solution: all scope provider information encoded in the grammar.
+    """
+
+    mm = metamodel_from_str('''
+        Model:
+            structs+=Struct
+            instances+=Instance
+            references+=Reference;
+        Struct:
+            'struct' name=ID '{' vals+=Val '}';
+        Val:
+            'val' name=ID (':' type=[Struct])?;
+        Instance:
+            'instance' name=ID (':' type=[Struct])?;
+        Reference:
+            'reference' ref=[Val|FQN|+p:instances.~type.vals.(~type.vals)*];
+        FQN: ID ('.' ID)*;
+        ''')
+    m = mm.model_from_str(model_text)
+    m.references[-1].ref == m.structs[0].vals[0]  # a.x
```

Comment:
This is already tested in line 307 bellow.

Reply:
(deleted)